python-checks/src/main/java/org/sonar/python/checks/FloatingPointEqualityCheck.java

```diff
@@ -0,0 +1,165 @@
+
+    context.registerSyntaxNodeConsumer(Tree.Kind.COMPARISON, this::checkFloatingPointEquality);
+  }
+
+  private void initializeAnalysis(SubscriptionContext ctx) {
+    reachingDefinitionsAnalysis = new ReachingDefinitionsAnalysis(ctx.pythonFile());
+    importedModuleForIsClose = null;
+    importedAlias = null;
+  }
+
+  private void checkFloatingPointEquality(SubscriptionContext ctx) {
+    BinaryExpression binaryExpression = (BinaryExpression) ctx.syntaxNode();
+    String operator = binaryExpression.operator().value();
+    if (("==".equals(operator) || "!=".equals(operator)) && isAnyOperandFloatingPoint(binaryExpression)) {
+      PreciseIssue issue = ctx.addIssue(binaryExpression, MESSAGE);
+      issue.addQuickFix(createQuickFix(binaryExpression, operator));
+    }
+  }
+
+  private boolean isAnyOperandFloatingPoint(BinaryExpression binaryExpression) {
+    Expression leftOperand = binaryExpression.leftOperand();
+    Expression rightOperand = binaryExpression.rightOperand();
+
+    return isFloat(leftOperand) || isFloat(rightOperand) ||
+        isAssignedFloat(leftOperand) || isAssignedFloat(rightOperand) ||
+        isBinaryOperationWithFloat(leftOperand) || isBinaryOperationWithFloat(rightOperand);
+  }
+
+  private static boolean isFloat(Expression expression) {
+    return expression.is(Tree.Kind.NUMERIC_LITERAL) && expression.type().equals(InferredTypes.FLOAT);
+  }
+
+  private boolean isAssignedFloat(Expression expression) {
+    if (expression.is(Tree.Kind.NAME)) {
+      Set<Expression> values = reachingDefinitionsAnalysis.valuesAtLocation((Name) expression);
+      if (!values.isEmpty()) {
+        return values.stream().allMatch(value -> isFloat(value));
+      }
+    }
+    return false;
+  }
+
+  private boolean isBinaryOperationWithFloat(Expression expression) {
+    if (expression.is(BINARY_OPERATION_KINDS)) {
+      return isAnyOperandFloatingPoint((BinaryExpression) expression);
+    }
+    return false;
+  }
+
+  private PythonQuickFix createQuickFix(BinaryExpression binaryExpression, String operator) {
+    String notToken = "!=".equals(operator) ? "!" : "";
+    String isCloseModuleName = getModuleNameOrAliasForIsClose();
+    String message = String.format(QUICK_FIX_MESSAGE, notToken, isCloseModuleName);
+    Builder quickFix = PythonQuickFix.newQuickFix(message);
+
+    String quickFixText = MATH_MODULE.equals(isCloseModuleName) ? QUICK_FIX_MATH : QUICK_FIX_IMPORTED_MODULE;
+    String quickFixTextWithModuleName = String.format(quickFixText, notToken, isCloseModuleName,
+        TreeUtils.treeToString(binaryExpression.leftOperand(), false),
+        TreeUtils.treeToString(binaryExpression.rightOperand(), false));
+
+    quickFix.addTextEdit(TextEditUtils.replace(binaryExpression, quickFixTextWithModuleName));
+
+    if (MATH_MODULE.equals(isCloseModuleName) && !isMathImported) {
+      quickFix.addTextEdit(TextEditUtils.insertAtPosition(0, 0, "import math\n"));
```

Comment:
One thing that may be annoying is that the user may not realize we're adding this import (the focus will not move as the quick fix gets added), and this may mess with stuff like license headers.

I don't really have a better solution: adding the import right before the `isclose` call may be better at getting the user attention (and then they can just move the import to where it belongs), but it's not ideal either to be honest.

Reply:
I agree this is not ideal. I am not sure as well how we could tackle this. Finding the first global import and inserting before or finding the last header line.